src/engine/tt_hash_table.h

```diff
@@ -14,20 +15,87 @@ enum TtHashFlag : uint8_t {
     TtHashBeta = 2,
 };
 
+/*
+ * Non-key members are encoded for compression. From lower bits to higher, entries are:
+ * depth: 7 bits, i.e. 0000 ... 0000 0000 0111 1111
+ * flag: 3 bits, i.e. 0000 ... 0000 0011 1000 0000
+ * score: 17 bits (see board_defs.h)
+ * generation: 8 bits
+ * move: 26 bits (see move_types.h)
+ * */
+
+class TtHashEntryData {
+public:
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wshift-count-overflow"
+    explicit TtHashEntryData(uint64_t depth = 0, uint64_t flag = TtHashExact, int64_t score = 0, uint64_t generation = 0, uint64_t move = {})
+        : m_data(
+              depth
+              | flag << s_flagShift
+              | score << s_scoreShift
+              | generation << s_generationShift
+              | move << s_moveShift)
+    {
+    }
+#pragma GCC diagnostic pop
+
+    constexpr inline uint8_t getDepth() const
+    {
+        return m_data & s_depthMask;
+    }
+
+    constexpr inline auto getFlag() const
+    {
+        return static_cast<TtHashFlag>((m_data >> s_flagShift) & s_flagMask);
+    }
+
+    constexpr inline int32_t getScore() const
+    {
+        return (m_data >> s_scoreShift) & s_scoreMask;
+    }
+
+    constexpr inline uint8_t getGeneration() const
+    {
+        return (m_data >> s_generationShift) & s_generationMask;
+    }
+
+    constexpr inline uint32_t getMoveData() const
+    {
+        return (m_data >> s_moveShift) & s_moveMask;
+    }
+
+private:
+    constexpr static inline uint64_t s_depthMask { 0b1111111 };
+
+    constexpr static inline uint64_t s_flagMask { 0b111 };
+    constexpr static inline uint64_t s_flagShift { 7 };
+
+    constexpr static inline uint64_t s_scoreMask { 0b11111111111111111 };
+    constexpr static inline uint64_t s_scoreShift { 10 };
+
+    constexpr static inline uint64_t s_generationMask { 0b11111111 };
+    constexpr static inline uint64_t s_generationShift { 27 };
+
+    constexpr static inline uint64_t s_moveMask { 0b11111111111111111111111111 };
+    constexpr static inline uint64_t s_moveShift { 35 };
+
+    uint64_t m_data;
```

Comment:
For default construction
```suggestion
    uint64_t m_data {};
```

Reply:
Good spot, thanks.